tests/components/insteon/test_config_flow.py

```diff
@@ -0,0 +1,703 @@
+    assert hass.config_entries.async_entries(DOMAIN)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        assert entry.data[CONF_HOST] == MOCK_HOSTNAME
+        assert entry.data[CONF_PORT] == 9761
+        assert entry.data[CONF_HUB_VERSION] == 1
+
+
+async def test_import_existing(hass: HomeAssistantType):
+    """Test we fail on an existing config imported."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        entry_id="abcde12345",
+        data={**MOCK_USER_INPUT_HUB_V2, CONF_HUB_VERSION: 2},
+        options={},
+    )
+    config_entry.add_to_hass(hass)
+    assert config_entry.state == config_entries.ENTRY_STATE_NOT_LOADED
+
+    result = await _import_config(hass, MOCK_IMPORT_MINIMUM_HUB_V2)
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_import_failed_connection(hass: HomeAssistantType):
+    """Test a failed connection on import."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    with patch(PATCH_CONNECTION, new=mock_failed_connection,), patch(
+        PATCH_ASYNC_SETUP, return_value=True
+    ), patch(PATCH_ASYNC_SETUP_ENTRY, return_value=True):
+        result = await async_import_config(hass, MOCK_IMPORT_MINIMUM_HUB_V2)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "cannot_connect"
+
+
+async def _options_init_form(hass, entry_id, step):
+    """Run the init options form."""
+    with patch(PATCH_ASYNC_SETUP_ENTRY, return_value=True):
+        result = await hass.config_entries.options.async_init(entry_id)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "init"
+
+    result2 = await hass.config_entries.options.async_configure(
+        result["flow_id"], {step: True},
+    )
+    return result2
+
+
+async def _options_form(hass, flow_id, user_input):
+    """Test an options form."""
+
+    with patch(PATCH_ASYNC_SETUP_ENTRY, return_value=True) as mock_setup_entry:
+        result = await hass.config_entries.options.async_configure(flow_id, user_input)
+        return result, mock_setup_entry
+
+
+async def test_options_change_hub_config(hass: HomeAssistantType):
+    """Test changing Hub v2 config."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        entry_id="abcde12345",
+        data={**MOCK_USER_INPUT_HUB_V2, CONF_HUB_VERSION: 2},
+        options={},
+    )
+
+    config_entry.add_to_hass(hass)
+    result = await _options_init_form(
+        hass, config_entry.entry_id, STEP_CHANGE_HUB_CONFIG
+    )
+
+    user_input = {
+        CONF_HOST: "2.3.4.5",
+        CONF_PORT: 9999,
+        CONF_USERNAME: "new username",
+        CONF_PASSWORD: "new password",
+    }
+    result, _ = await _options_form(hass, result["flow_id"], user_input)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)